### PR TITLE
Add more shirt stats

### PR DESCRIPTION
**Pull request by willgearty**
This adds t-shirt statistics to the dashboard for the following groups:

- All teachers
- Enrolled students
- Attended students
- Volunteers

Fixes https://github.com/learning-unlimited/ESP-Website/issues/2981 and fixes https://github.com/learning-unlimited/ESP-Website/issues/126.

### CODE REVIEW

File: esp/esp/program/models/__init__.py [hunk 1]
```diff
@@ -1151,38 +1151,95 @@ def getVolunteerRequests(self):
...
+
         teacher_dict = self.teachers()
-        if 'class_approved' in teacher_dict:
-            query = teacher_dict['class_approved']
-            query = query.filter(registrationprofile__most_recent_profile=True)
-            if not Tag.getBooleanTag('teacherinfo_shirt_type_selection', default=True):
-                query = query.values_list('registrationprofile__teacher_info__shirt_size')
-                query = query.annotate(people=Count('id', distinct=True))
+        teacher_types = {'Approved Teachers': 'class_approved', 'All Teachers': 'class_submitted'}
+        shirt_sizes = [x.strip() for x in Tag.getTag('teacher_shirt_sizes', default = 'XS, S, M, L, XL, XXL').split(',')]
```
Comment (hwatheod) [inline esp/esp/program/models/__init__.py:1159]: Now that there is a global tagdict with the default values for each tag, is there a way to write this that doesn't involve specifying the default value?

Comment (willgearty) [reply esp/esp/program/models/__init__.py:1159]: My plan is to make a separate change to `getTag` for that in a separate PR ;)

Comment (hwatheod) [reply esp/esp/program/models/__init__.py:1159]: ok, sounds good.

**Review by hwatheod — COMMENTED** [review]
It looks like mostly the same code is repeated 3 times for t-shirts with students, teachers, volunteers. Can you put this into a function?

**Review by hwatheod — APPROVED** [review]

**Pull request merged**